src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -83,10 +83,26 @@ enum ResolutionFailure<'a> {
     /// this could be an enum variant, but the last path fragment wasn't resolved.
     /// the `String` is the variant that didn't exist
     NotAVariant(Res, Symbol),
+    /// This link has malformed generic parameters; e.g., the angle brackets are unbalanced.
+    MalformedGenerics(MalformedGenericsVariant),
     /// used to communicate that this should be ignored, but shouldn't be reported to the user
     Dummy,
 }
 
+#[derive(Debug)]
+enum MalformedGenericsVariant {
```

Comment:
nit: this is a little verbose, maybe just `MalformedGenerics`?

Reply:
I would prefer that but I think it will be confusing because there are some cases of `use ResolutionFailure::*` which will shadow the variant.

Reply:
I'd prefer removing `use *` to having the longer name I think. Or you could match on `ResolutionFailure::MalformedGenerics` which will let you keep the star imports.

Reply:
I think this comment got lost in the shuffle.

Reply:
Yeah, 54 commits and counting! The GitHub UI has been kind of glitchy for me on this PR; I think it's overwhelmed :)